src/components/Chart/renderSeries.tsx

```diff
@@ -0,0 +1,212 @@
+      strokeWidth={2}
+      yAxisId='data'
+      connectNulls
+      label={<SeriesLabel seriesType={SeriesType.Line} color={color} />}
+      strokeDasharray={dashed && '6 4'}
+      legendType={dot === false ? 'plainline' : (dot as LegendType) ?? 'circle'}
+    />
+  )
+}
+
+function renderArea<XDomain>(
+  index: number,
+  series: ChartSeries<XDomain>,
+  showTooltip: boolean,
+  xDomain: AxisDomain,
+  stacked: boolean,
+  colorScheme: ChartColorScheme,
+  tooltipRenderer?: TooltipRenderer<XDomain>
+) {
+  const { name, dashed, dot } = series
+  const cs = getChartColorScheme(colorScheme)
+  const color = series.color ?? cs[index % cs.length]
+
+  return (
+    <Area
+      {...getDefaultRenderProps(name, color)}
+      fillOpacity={0.1}
+      dot={
+        dot === false ? (
+          false
+        ) : (
+          <LineDot showTooltip={showTooltip} xDomain={xDomain} tooltipRenderer={tooltipRenderer} dotShape={dot} />
+        )
+      }
+      strokeWidth={2}
+      yAxisId='data'
+      connectNulls
+      label={<SeriesLabel seriesType={SeriesType.Line} color={color} />}
+      strokeDasharray={dashed && '6 4'}
+      legendType={dot === false ? 'plainline' : (dot as LegendType) ?? 'circle'}
+      stackId={stacked ? 'stackId' : undefined}
+    />
+  )
+}
+
+function renderColumn<XDomain>(
+  index: number,
+  series: ChartSeries<XDomain>,
+  stacked: boolean,
+  showTooltip: boolean,
+  xDomain: AxisDomain,
+  colorScheme: ChartColorScheme,
+  tooltipRenderer: TooltipRenderer<XDomain>
+) {
+  const { name } = series
+  const cs = getChartColorScheme(colorScheme)
+  const color = series.color ?? cs[index % cs.length]
+
+  return (
+    <Bar
+      {...getDefaultRenderProps(name, color)}
+      stackId={stacked && 'stackId'}
+      yAxisId='data'
+      shape={(props) => (
+        <ChartTooltip
+          labelDomain={xDomain}
+          label={(props as any).payload.x}
+          value={(props as any).value}
+          showTooltip={showTooltip}
+          renderer={tooltipRenderer}
+        >
+          <Rectangle {...props} />
+        </ChartTooltip>
+      )}
+      label={{ fill: color, position: 'top' }}
+      legendType='circle'
+    >
+      {stacked && <LabelList dataKey={name} position='insideTop' offset={12} {...{ fill: 'white' }} />}
+    </Bar>
+  )
+}
+
+export function renderBar<YDomain>(
+  index: number,
+  series: BarChartSeries<YDomain>,
+  stacked: boolean,
+  showTooltip: boolean,
+  yDomain: AxisDomain,
+  colorScheme: ChartColorScheme,
+  tooltipRenderer?: TooltipRenderer<YDomain>
+) {
+  const { name } = series
+  const cs = getChartColorScheme(colorScheme)
+  const color = series.color ?? cs[index % cs.length]
+
+  return (
+    <Bar
+      {...getDefaultRenderProps(name, color)}
+      stackId={stacked && 'stackId'}
+      yAxisId='data'
+      shape={(props) => (
+        <ChartTooltip
+          labelDomain={yDomain}
+          label={(props as any).payload.x}
+          value={(props as any).value}
```

Comment:
Can this `props` be casted to another type ?

Reply:
The lib is not very consistent with its own documentation, so I needed to force some casts to comply with what's really happening when running the code. Anyway, I added a new interface to represent the type of these props.

Reply:
Okayy